lib/declaration.ts

```diff
@@ -963,14 +968,32 @@ export default class DeclarationTranspiler extends base.TranspilerBase {
     this.emitNoSpace(';');
   }
 
-  private emitMethodsAsExtensions(
-      className: ts.Identifier, visitName: () => void, visitNameOfExtensions: () => void,
+  private emitMembersAsExtensions(
+      className: ts.Identifier, visitClassName: () => void, visitNameOfExtensions: () => void,
       methods: ts.SignatureDeclaration[]) {
+    const visitTypeAndArguments = (type: ts.TypeNode) => {
+      if (ts.isTypeReferenceNode(type) && base.ident(type.typeName) === 'Promise') {
+        this.emit('Future');
+        if (ts.isTypeReferenceNode(type)) {
+          this.maybeVisitTypeArguments(type);
+        }
+      } else {
+        this.fc.visit(type);
+      }
+    };
+
+    const castThisToPrivateClass = () => {
```

Comment:
why isn't this a method? based on the usage it can just be a method on this class.